node/core/pvf/src/error.rs

```diff
@@ -79,16 +98,20 @@ impl fmt::Display for PrepareError {
 pub enum ValidationError {
 	/// The error was raised because the candidate is invalid.
 	InvalidCandidate(InvalidCandidate),
-	/// This error is raised due to inability to serve the request.
-	InternalError(String),
+	/// This error is raised due to inability to serve the request during preparation.
+	InternalPrepareError(NonDeterministicError),
```

Comment:
Why not forward the prepareerror directly?

The variants also don't need the postfix "Error".

Reply:
And yeah, looking into the code that handles this error, we should name this variant `Prepare` and forward the `PrepareError` directly.

Reply:
These internal variants are only for non-deterministic errors, hence the `Internal` prefix and why I use `NonDeterministicError` here.

Reply:
Ahh, I had read the From implementation not good enough! Sorry!